.github/workflows/continuous-integration.yml

```diff
@@ -1,96 +1,77 @@
-        uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
-      - uses: actions/setup-java@v1
-        with:
-          java-version: '11'
-      - name: Build and Test ${{ env.PACKAGE_NAME }}
+      - name: Setup Environment
         run: |
-          python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
-          chmod a+x builder.pyz
-          AWS_CRT_SWIFT_CI_DIR="${{ env.AWS_CRT_SWIFT_CI_DIR }}" AWS_SDK_SWIFT_CI_DIR="${{ env.AWS_SDK_SWIFT_CI_DIR }}" SMITHY_SWIFT_CI_DIR="${{ env.SMITHY_SWIFT_CI_DIR }}" ./builder.pyz build -p ${{ env.PACKAGE_NAME }} --target=ios-armv8
-  macos-compat:
-    runs-on: macos-11
-    env:
-      DEVELOPER_DIR: /Applications/Xcode_13.2.1.app
-    steps:
-      - name: Checkout sources
+          echo "AWS_SDK_REF_TO_BUILD=$GITHUB_REF_NAME" >> $GITHUB_ENV
+          echo "AWS_SDK_XCODEBUILD_DESTINATION=${{ matrix.destination }}" >> $GITHUB_ENV
+      - name: Checkout Sources
         uses: actions/checkout@v2
-      - uses: actions/cache@v2
+      - name: Setup Java
+        uses: actions/setup-java@v1
         with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
-      - uses: actions/setup-java@v1
-        with:
-          java-version: '11'
-      - name: Build and Test ${{ env.PACKAGE_NAME }}
-        run: |
-          python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
-          chmod a+x builder.pyz
-          AWS_CRT_SWIFT_CI_DIR="${{ env.AWS_CRT_SWIFT_CI_DIR }}" AWS_SDK_SWIFT_CI_DIR="${{ env.AWS_SDK_SWIFT_CI_DIR }}" SMITHY_SWIFT_CI_DIR="${{ env.SMITHY_SWIFT_CI_DIR }}" ./builder.pyz build -p ${{ env.PACKAGE_NAME }}
+          java-version: 18
+      - name: Set References
+        run: ./scripts/steps/set_references.sh
+      - name: Install Dependencies
+        run: ./scripts/steps/install_dependencies.sh
+      - name: Perform Test Codegen
+        run: ./scripts/steps/perform_test_codegen.sh
+      - name: Post-Codegen Cleanup
+        run: ./scripts/steps/post_codegen_cleanup.sh
+      - name: Build and Test With xcodebuild
+        run: ./scripts/steps/build_and_test_with_xcodebuild.sh
   linux-compat:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        image:
-          - ubuntu-x64
-          - al2-x64
+        os:
+          - jammy
+          - amazonlinux2
```

Comment:
could you add different archs here also
ie x86 and ARM

Reply:
I will have to see if Github has ARM Linux executors.  If they do, I will add

Reply:
They do not.  We would have to self-host CI runners to build on ARM.